### DISCUSSION TITLE
Welcome to ScoutingPASS Discussions!

### DISCUSSION THREAD

**Discussion by fuddster** [PWNAGERobotics/ScoutingPASS#20] [Announcements]
## 👋 Welcome to ScoutingPASS Discussions

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re about our App
  * Share ideas on how to improve it
  * Submit bugs (although we would prefer you open an issue for it)
  * Engage with other community members.

We'd like to get to know you and what brought you to our App.  

Please add a comment with your FRC team name and number.  We'd love to know if your planning on using ScoutingPASS or just curious.

Thank you!

PWNAGE

**Comment by FRC3931** [comment]
I am the scouting  Mentor for our team. I am very interested in using your scouting software. I am not a programmer but I’ve used VBA inside excel a whole lot. I know just enough about JavaScript to be dangerous. I have never seen GitHub before. I tried to follow your instructions and also some other things that I saw on GitHub. Right now I can’t get the main thing to show up in our repository. I also cannot figure out how to copy the software to our depository. I am sure I will have more questions later but for now I would really like to just play with it. I appreciate any help you can offer in advance. Thank you.

**Reply by fuddster** [reply]
So, where you want to start is to click on "Fork" up in the top right corner of this page.  Answer the questions and then you'll have a copy of the ScoutingPASS software in your account.  It will be under repositories.

Now go to your copy and you can view all the code and configurations.   Near the top are a bunch of "tabs": Code, Issues, Pull Requests, etc.   At the end is Settings.  Click on Settings.   Then on the lefthand menu click on "Pages".   Under source, select the Main branch and click Save.   It will display a message that says your site is live and give you the URL.    You'll probably have to wait 5-15 minutes for the page to populate, but it should be there eventually.

Once you're up on your own site, you can play around with the configuration and all.

Let me know how it goes.

Scott Elmer
Scouting Mentor
PWNAGE #2451

**Comment by Evan-FRC** [comment]
Hi,
My team and I have enjoyed our experience using ScoutingPASS so far, but we've encountered an issue where the text and background of the input boxes on the pre-match page are both the same color, making it unable for us to see what we are typing. For some reason, this issue occurs only on the Samsung A7 lite tablets that we've been using for our scouting system and on the Google browser. If we switch to other devices or other browsers on the tablet, the background is able to be seen. Has your team had any issues with this or do you know any ways this can be fixed?

Thanks for your help,
Evan

**Reply by fuddster** [reply]
Hi Evan.

We haven't noticed anything like that in our experience, but we don't use tablets.    You could try changing the offending colors and see if that helps.   We've recently updated the app including improving the color scheme.  Try using the latest code and see if that works.   

If not, try changing the color in this line in index.html:
`<div id="prematch" class="main-panel" style="background-color: black;">`
Use another darker color or else the text won't show up.

Let me know how it goes.

Scott

**Comment by FRC3931** [comment]
It appears to have worked. Thank you. I don't know what I did or didn't do
the last time. I am not sure I care.

So now is there some kind of relationship between my version and yours?

I know I will need to know these things but am not sure how to ask them.
1. What is the name of each variable (or field or whatever) from each entry
on the tablet?
2. What is the expected output from radio buttons? (R1-3, B1-3 for example)
3. In the description you said I could pick from certain data types? If I
wanted to add or change things how do I set the data type?
4. I think the description said that I could prepopulate team numbers from
the match schedule. Can I assign a certain tablet to be R1, remove the
choice for that question, and then populate the teams in that spot? It
seems like I would need six different files and have R1 download that
specific file.
5. Where is the field image file with the grids? If we added that for
shooting locations and robot A shot from 10 different locations what would
that output look like? I don't suppose you can do make versus miss on the
same image field?
6. Can the tablets be Kindle Fire? Do you know if there are any additional
software things needed to make this work on that? We have done this before
with a different system that didn't work out too well. I am not sure if the
software just had a lot of issues or if this software emulator thing was
part of the problem.

Thank you so much for your help. I am sure I will be in touch with more
questions as we get into this.

Eddie Shearer

**Reply by fuddster** [reply]
Eddie,

Glad you got it working.  Let me see if we can get some of these questions answered:

1. Look in the configuration file (directory: 2022/RR_config.js).   Each field has a "code" associated with it.   That code is what is used in the QR code to transfer the data.
2. For radio buttons you can configure the output however you'd like.   For the "robot" field (code='r'), the choices are:
          "r1":"Red-1",
          "b1":"Blue-1",
          "r2":"Red-2",
          "b2":"Blue-2",
          "r3":"Red-3",
          "b3":"Blue-3"
If you pick Red-1 then r1 will be returned.   In the QR code you will see r=r1, if you pick Blue-3, you'll see r=b3.
3. The data types are fixed.  You can see the list of them in the README.md.  Your programmers could add more if you're so inclined.
4. You can't assign a certain tablet to a specific robot.  However, once you pick the robot to scout, it will stay on that robot unless specifically changed.   The "Clear Form" button increases the match number and keeps the same robot.  So, it's not foolproof, but it's something.
5. Look at the FAQ in the README.md for the Field Image question.   It describes how the field image element works and shows the grid for the mapping of the numbers.  Here's how we used the data in 2020 to graph it in Excel.  It required some VBA programming and beyond the scope of this application. ![ShootingGrid](https://user-images.githubusercontent.com/8154245/156273670-a1b6eefc-e98b-4d05-abed-c608f72c7b6f.png)
The hit vs miss is an interesting idea.   That's not currently a feature.
6. It should work on anything that has a web browser.  I'd try it before competitions though to make sure.

Good luck.

Scott

**Comment by FRC3931** [comment]
I tried to modify the configuration. I don't know if I did it right or not.
I cannot view my version of the app (FRC3931.github.io/ScoutingPASS). Each
time I get a 404 error.

Do I need to modify the Excel file (some save data line) in the same way as
I modified the configuration? Does order matter on the scanner portion?
Where will it save the data to? Does that save data line refer to default
values or data types equal to the field name?

Thanks again for your help.

Eddie Shearer

**Reply by fuddster** [reply]
Hi,

First off it looks like you haven't setup the GitHub pages. 

Did you do this step?

* Enable GitHub Pages in your repository settings (GitHub Pages Instructions)

Basically go into the repository settings, select "Pages".  On the next page select "main" branch and click save.  It should display a URL, the same one you showed me.  It will take up to 10 min or so for the pages to be deployed. After, 10 min try the URL and it should work.

**Comment by barkinvar** [comment]
Hi is it possible to change the size of the field image it is hard to press on smaller devices and also is there a way to increase the rendered resolution?

**Comment by landaj16** [comment]
Just to clarify...
Is it possible to use a qr scanner device to send your data to a Google sheet OR if a qr scanner device is used will it only pull data into excel?

**Reply by fuddster** [reply]
The QR code and sending to Google Sheets are separate functions. 

The QR scanner is connected to a laptop and needs something to read the input.  In this case, it's Excel.  

The Google Sheets functionality is a button on the QR screen that sends the data directly to your Google Sheet over the internet.

I guess it's possible to write a script for Google Sheets to take an input from a QR code, but since you have the ability to send it straight to the sheet I'm not sure why you would want to put the scanner in the middle of that interaction.

**Comment by landaj16** [comment]
Basically everyone seems to prefer Google sheets over excel and everyone
wants to use the scanner bc of wifi issues at events so they were thinking
we could use the scanner and it would go to a Google sheet instead of
excel. That's the thought process the head mentors had.

**Reply by fuddster** [reply]
I can see why some teams would want to use Google Sheets.  However, I haven't run across many that want to use it offline.

Thank you for explaining your use case and we'll see if that's something we can add in a future release.

**Comment by rlr244** [comment]
Hi! Team 4188 is excited to use this system for scouting at our first event this weekend. I forked it and had the TBA API working great with the sample event from 2020, but when I put in event codes from this season week 1, I am not seeing any teams returned on the pre-match screen. Am I missing something in the event code? thanks!

**Reply by fuddster** [reply]
Welcome to ScoutingPASS!

It won't work until the official match schedule is released.  Which usually isn't until the night before or even the morning that qualification matches start. 

What event?  I will check it out.

**Reply by rlr244** [reply]
Thanks for the quick reply! I've tried it with ISR District Event #2 (ISDE2) and Canadian Pacific Regional (BCVI) which are both underway. Maybe I'm missing exactly where to put it, I'm just replacing the event default value, line 19 in the CU_config file

**Reply by fuddster** [reply]
Let me check…

**Reply by fuddster** [reply]
It seems to be working on my fork.   Did you include the year?   That’s part of the event code: 2023bcvi

**Reply by rlr244** [reply]
Thanks so much! I think it was a combo of year and case sensitivity. I appreciate your help! We are now ready for Peachtree District Albany event tomorrow!

**Reply by fuddster** [reply]
Good luck!  PWNAGE will be rooting for you!

**Reply by landaj16** [reply]
Do we only have to update the event code in order to get the api for an event to work. i.e I am looking to test out the canadian pacific regional tomorrow even though we will not be attending.  Not really sure I am understanding how to request the token and get the access code and then add it into the tba interface code.

**Reply by fuddster** [reply]
Yes, the event code change is all that’s needed.   However, we have exposed a TBA key that we consider “public”.  You can use it, but if it gets abused (thousands of requests a day) it could be deactivated and the TBA integration will no longer work.  

In order to protect against this we HIGHLY recommend that you get your own key for competitions.

**Comment by PaotheCrayo** [comment]
Hi! I'm attempt to make a Google Sheet for my team's scouting, unfortunately, I got a little lost after the "**Scouting P.A.S.S Set Up Set up the URL.** In resources/js/googleSheets.js, on line 2, replace <SCRIPT URL> with the URL you just copied from Google Sheets. It needs to stay wrapped in single quotes." part for your ScoutingPASS app. Is this a file I need to download, or is this a file I should have already? Any and all advice is greatly appreciated. Thank you so much for such an amazing resource!

**Reply by fuddster** [reply]
Hi.  Thanks for choosing ScoutingPASS!

So, the Google Sheet integration is done in two parts.  

Part 1 - The web app with the screens to enter scouting data and format the data for Google Sheets
Part 2 - The google sheet and the script to parse the data

The glue between those two parts is a URL. (i.e. https:// script.google.com/macros/s/REALLY LONG STRING OF NUMBERS/exec)
You get this URL from the deployment of the Google Apps Script.  (See Steps 7 and 8 in the Google Sheets Setup section)

If you missed copying that URL when you deployed the script, you could find it again by following these steps:

1. Go to your Google Sheets spreadsheet
2. Select the Extensions menu item and choose Apps Scripts.  It will pop open a new browser window/tab with your script in it.
3. In the new Apps Script window select the Blue "Deploy" button in the upper right corner and select "Manage Deployments."  It will open a pop-up with your deployments.  The latest deployment should already be selected.
4. Under "Web App" there will be a URL.  Click "Copy" to copy the URL into your clipboard
5. Now go back to your GitHub code and the googleScripts.js script and replace <SCRIPT URL> in line 2 with the URL you just copied.

That should get you past that step. 

There was a bug fix to the Google Sheets script last week.  Make sure you have the latest version (remove .join(',') from line 22.)

If it's still not working contact me with your repository link and the Google Sheets link and I'll take a look at it for you.

**Comment by rlr244** [comment]
We had great success using ScoutingPass for the PCH Albany event! With it being a week 1 event, our scouting team didn't have much training, and they were still able to enter solid data. I received feedback from the scouts asking if the grid positions selected in Auto can carry over to the TeleOp portion, since those game pieces are still there. Any ideas how to carry these forward?

**Reply by fuddster** [reply]
That's an interesting thought.  The way it's written right now there's no way to carry it over.   They are two separate data inputs and no way to connect them.   

But that's something we can keep in mind as an improvement for the future.

**Comment by landaj16** [comment]
I am going to try and request my own key, but once I get it where do I post
our new key?

**Reply by landaj16** [reply]
I requested a read api on blue alliance like a week ago and got something that looks like this


X-TBA-Auth-Key | Created | Description |  
-- | -- | -- | --
o3G4X0ms8efdV1nHUlbNTfaNufACaRS1V3VBGrrdWGGHTd40U3tmwvLzMnuSlMOC | 2023-03-03 | Scouting App | Delete


<br class="Apple-interchange-newline">

**Reply by fuddster** [reply]
Great!   Now plug that into the TBAInterface.js file.   Go to resources/js/TBAInterface.js and replace the long string on line 4 with the string above and you should be good.

**Comment by landaj16** [comment]
Ok so that in addition to the name change of the event? Also I'm looking
for the read not the write api correct?

**Reply by fuddster** [reply]
Yes, you need the READ api key.

**Reply by landaj16** [reply]
I have an event this weekend at finger lakes you think I can request that api token tonight or would I need to wait till say tomorrow night or Friday morning?

**Reply by landaj16** [reply]
Well I actually got this when I requested it for finger lakes 
Token:9adac43c-3554-4f62-8575-ea0bcdbb9ab1 not sure what to do with this to get authentification key can't remember how I got that last time-

**Reply by fuddster** [reply]
All you need is the Auth Key, which you posted above.    You don't need a token.

Then you need to update the even field with the event code.   For finger lakes it is 2023nyro.  (All lower case)  Put that in the configuration file, 2023/CU_config.js or 2023/CU_GS_config.js.

**Reply by landaj16** [reply]
I entered 2023nyro in both files to be safe that you mentioned above and it did not update to the finger lakes.  Would that be because the API is just not accessible yet.  All matches are loaded onto the blue alliance for tomorrow so I assumed they are accessible.

**Reply by fuddster** [reply]
Looks like you're good to go now.   I tried your web site and it's updating the team number and name.

If you aren't seeing it, try refreshing the web page.

Good luck this weekend!

**Comment by landaj16** [comment]
The other thing I posted was from a previous event but I never used it so
all I have for finger lakes is a token right now. I'm assuming I need to
request the api with this token to get that.
